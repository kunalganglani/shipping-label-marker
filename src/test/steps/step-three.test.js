import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import GetWeight from '../../features/shipping-label-maker/steps/step-three';

Enzyme.configure({ adapter: new Adapter() });
const wizardContext = {
  from: {
    name: '',
    street: '',
    city: '',
    state: '',
    zip: '',
  },
  to: {
    name: '',
    street: '',
    city: '',
    state: '',
    zip: '',
  },
  weight: {
    weightValue: '',
    weightUnit: 'Pounds',
  },
  shippingOption: '1',
};
test('GetWeight renders correctly', () => {
  const handleAction = jest.fn();
  const tree = renderer
    .create(<GetWeight wizardContext={wizardContext.weight} onAction={handleAction} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('createAction is called', () => {
  const someFunction = jest.fn();
  const wrapper = shallow(<GetWeight wizardContext={wizardContext.weight} onAction={someFunction} />);
  wrapper.find('WithStyles(ForwardRef(TextField))').first().simulate('change', {
    target: {
      value: 1,
    },
  });
  expect(someFunction).toHaveBeenCalled();
});

test('handleChange is called', () => {
  const someFunction = jest.fn();
  const wrapper = shallow(<GetWeight wizardContext={wizardContext.weight} onAction={someFunction} />);
  wrapper.find('WithStyles(ForwardRef(Select))').first().simulate('change', {
    target: {
      value: 'KG',
    },
  });
  expect(someFunction).toHaveBeenCalled();
});
