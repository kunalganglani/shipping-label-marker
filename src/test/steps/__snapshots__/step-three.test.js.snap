// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetWeight renders correctly 1`] = `
<div
  className="css-102a0lm"
>
  <h3>
    Enter Weight
  </h3>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="weightValue"
      id="weightValue-label"
    >
      Weight
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="weightValue"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="number"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-3"
        >
          <span>
            Weight
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiFormControl-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
      data-shrink={true}
      id="demo-simple-select-outlined-label"
    >
      Weight Type
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        aria-labelledby="get-weight-type-label weightUnit"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
        id="weightUnit"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        Pounds
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
        value="Pounds"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
        >
          <span>
            Weight Type
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
